### PR TITLE
Add moyo_list:position_if/2

### PR DESCRIPTION
**Pull request by gfngfn**
This PR attempts to add a new function `moyo_list:position_if/2`, which can be used in the form `moyo_list:position_if(PredFun, List)` to get the index of the first element in `List` that satisfies the predicate `PredFun` if exists, or to get `error` otherwise.

Although we can realize the same functionality as that of `moyo_list:position_if/2` by combining `moyo_list:find_if/2` and `moyo_list:position/2`, this implementation would be more efficient in that it does not require traversing the same list twice.

I will appreciate any comments or suggestions.

### CODE REVIEW

**Review by yaoshimax — APPROVED** [review]

**Review by zeek0x — COMMENTED** [review]
`moyo_list` についての edoc を更新して欲しいです :pray:

**Comment by gfngfn** [comment]
I updated `doc/moyo_list.md` simply by invoking `make edoc`.

Since many documents seem to have failed to keep up changes in `src/` for a long time, just invoking `make edoc` makes a somewhat large diff in `doc/`. It would be better to make another PR for updating the documents.

**Review by zeek0x — APPROVED** [review]

File: src/moyo_list.erl [hunk 1]
```diff
@@ -107,6 +108,13 @@ replace_if(PredFun, Value, List) -> replace_if_impl(PredFun, Value, List, []).
 -spec position(term(), list()) -> {ok, Position::pos_integer()} | error.
 position(Value, List) -> position_impl(Value, List, 1).
 
+%% @doc `List'内で最初に述語`PredFun'を満たす要素の位置を返す
+%%
+%% 該当する要素が存在しない場合は`error'が返される
```
Comment (koba-e964) [inline src/moyo_list.erl:113]: Could you mention that the returned index starts at 1, like in the parameter of `lists:nth/2`?

Comment (gfngfn) [reply src/moyo_list.erl:113]: I added explanation about that (although `pos_integer()` in the spec indicates to some extent that return values are 1-origin). Thanks for suggestion.

**Review by koba-e964 — APPROVED** [review]

**Pull request merged**